pkg/didcomm/crypto/authcrypt/encrypt_test.go

```diff
@@ -0,0 +1,71 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package authcrypt
+
+import (
+	"bytes"
+	"crypto/ecdsa"
+	"crypto/elliptic"
+	"crypto/rand"
+	"encoding/json"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestEncrypt(t *testing.T) {
+	sendEcKey, err := ecdsa.GenerateKey(elliptic.P256(), rand.Reader)
+	require.NoError(t, err)
+
+	recipient1Key, err := ecdsa.GenerateKey(elliptic.P256(), rand.Reader)
+	require.NoError(t, err)
+
+	recipient2Key, err := ecdsa.GenerateKey(elliptic.P256(), rand.Reader)
+	require.NoError(t, err)
+
+	recipient3Key, err := ecdsa.GenerateKey(elliptic.P256(), rand.Reader)
+	require.NoError(t, err)
+
+	_, err = NewAuthCrypter(sendEcKey, []interface{}{&recipient1Key.PublicKey, &recipient2Key.PublicKey, &recipient3Key.PublicKey}, "BAD")
+	require.Error(t, err)
+	badKey := ""
+	_, err = NewAuthCrypter(badKey, []interface{}{&recipient1Key.PublicKey, &recipient2Key.PublicKey, &recipient3Key.PublicKey}, "XC20P")
```

Comment:
Let's break these test cases into separate tests such that the focus of the test becomes clearer. Right now it's rather hard to see what exactly is being tested in each assertion.

Reply:
will separate them, or add comment for each test case, probably not worth creating a test function for each test case

Reply:
added comments for each test case

Reply:
If we're going to stick all scenarios into a single method then I'd prefer we have them like in [this example](https://github.com/hyperledger/aries-framework-go/blob/4380aaed9d8282b324bb9d69a904b0888ab5d28b/pkg/framework/didresolver/didresolver_test.go#L66), which has several virtues:

* aesthetics: much cleaner separation of test cases
* each anonymous function declares its own variables which reside in the anonymous scope. This avoids coupling between tests

Reply:
will update..

Reply:
done